// react-native-gifted-chat/src/components/SliderEntry.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, TouchableWithoutFeedback, InlineImage } from 'react-native';
import PropTypes from 'prop-types';
import { ParallaxImage } from 'react-native-snap-carousel';
import styles from '/Users/catherinepricillas/Chatbot/node_modules/react-native-gifted-chat/src/styles/SliderEntry.style';

export default class SliderEntry extends Component {

      constructor(props){
        super(props);
      }

    static propTypes = {
        data: PropTypes.object.isRequired,
        onPress: PropTypes.func.isRequired,
        even: PropTypes.bool,
        parallax: PropTypes.bool,
        parallaxProps: PropTypes.object
    };

  _onPress(text2) {
      this.props.onPress(text2);
  }

  _onPress2() {
      this.props.onPress2();
  }

  _onPress3(text2, idx) {
      this.props.onPress3(text2, idx);
  }


    get image () {
        const { data: { illustration }, parallax, parallaxProps, even, idx } = this.props;

        return parallax ? (
            <ParallaxImage
              source={illustration}
              containerStyle={[styles.imageContainer, even ? styles.imageContainerEven : {}]}
              style={styles.image}
              parallaxFactor={0}
              showSpinner={true}
              spinnerColor={even ? 'rgba(255, 255, 255, 0.4)' : 'rgba(0, 0, 0, 0.25)'}
              {...parallaxProps}
            />
        ) : (
            <Image
              source={illustration}
              style={styles.image}
            />
        );
    }


    render () {
        const { data: { title, subtitle }, even, idx } = this.props;

        const uppercaseTitle = title ? (
            <Text
              style={[styles.title, even ? styles.titleEven : {}]}
              numberOfLines={2}
            >
                { title }
            </Text>
        ) : false;

        var icon = this.props.data.love ? require('/Users/catherinepricillas/Chatbot/shared/static/heartlined.png') : require('/Users/catherinepricillas/Chatbot/shared/static/heartout.png');

        if (this.props.data.next === true) {
          return (
            <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              onPress={()=>this._onPress(`${this.props.data.title}`)}
              >
                <View style={styles.shadow} />
                <View style={styles.imageContainer} >
                    <Image
                      source={this.props.data.illustration}
                      style={styles.image2}
                    />
                    <View style={styles.radiusMask} />
                </View>
                <View style={styles.textContainerButton}>
                </View>
            </TouchableOpacity>
          );
        }
        return (
            <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              onPress={()=>this._onPress2()}
              >
                <View style={styles.shadow} />
                <View style={[styles.imageContainer, even ? styles.imageContainerEven : {}]} >
                    { this.image }
                    <View style={[styles.radiusMask, even ? styles.radiusMaskEven : {}]} />
                </View>
                <View>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress3(`Tambahkan ${this.props.data.title} ke wishlist`, {idx})}>
                    <Image style={{height: 31, width: 35, marginTop:-140, marginLeft:220}}
                        source={icon}
                      />
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
                <View style={[styles.textContainer, even ? styles.textContainerEven : {}]}>
                    { uppercaseTitle }
                    <Text
                      style={[styles.subtitle, even ? styles.subtitleEven : {}]}
                      numberOfLines={2}
                    >
                        { subtitle }{'                         '}
                    <Image style={{height: 14, width: 60}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/five-stars.png')}
                      />
                    </Text>
                </View>
                <View style={styles.textContainer}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress2()}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Lihat Detail Produk
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
                <View style={styles.textContainer}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Produk serupa ${this.props.data.title}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Produk Serupa
                    </Text>
                    </TouchableOpacity>

                    </TouchableWithoutFeedback>
                </View>
                <View style={styles.textContainerButton}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Beli ${this.props.data.title}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Beli Produk
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableOpacity>
        );
    }
}
