// react-native-gifted-chat/src/components/SliderEntry4.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, TouchableWithoutFeedback, InlineImage } from 'react-native';
import PropTypes from 'prop-types';
import { ParallaxImage } from 'react-native-snap-carousel';
import styles from '/Users/catherinepricillas/Chatbot/node_modules/react-native-gifted-chat/src/styles/History.style';

export default class SliderEntry4 extends Component {

      constructor(props){
        super(props);
      }

    static propTypes = {
        data: PropTypes.object.isRequired,
        onPress: PropTypes.func.isRequired,
        even: PropTypes.bool,
        parallax: PropTypes.bool,
        parallaxProps: PropTypes.object
    };

  _onPress(text2) {
      this.props.onPress(text2);
  }

  _onPress2() {
      this.props.onPress2();
  }

  _onPress3(text2, idx) {
      this.props.onPress3(text2, idx);
  }


    get image () {
        const { data: { illustration }, parallax, parallaxProps, even, idx } = this.props;

        return parallax ? (
            <ParallaxImage
              source={illustration}
              containerStyle={[styles.imageContainer, even ? styles.imageContainerEven : {}]}
              style={styles.image}
              parallaxFactor={0}
              showSpinner={true}
              spinnerColor={even ? 'rgba(255, 255, 255, 0.4)' : 'rgba(0, 0, 0, 0.25)'}
              {...parallaxProps}
            />
        ) : (
            <Image
              source={illustration}
              style={styles.image}
            />
        );
    }


    render () {
        const { data: { title, subtitle }, even, idx } = this.props;

        const uppercaseTitle = title ? (
            <Text
              style={[styles.title, even ? styles.titleEven : {}]}
              numberOfLines={2}
            >
                { title }
            </Text>
        ) : false;

        var icon = this.props.data.love ? require('/Users/catherinepricillas/Chatbot/shared/static/heartlined.png') : require('/Users/catherinepricillas/Chatbot/shared/static/heartout.png');

        return (
            <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              >
                <View style={styles.shadow} />
                <View style={[styles.imageContainer]} >
                    <Text
                      style={styles.titleEven}
                    >
                      Order: {this.props.data.subtitle}
                    </Text>
                    <Text
                      style={styles.title}
                    >
                      Tanggal: 22/05/2018
                    </Text>
                    <Text
                      style={styles.title}
                    >
                      1 x {this.props.data.title}
                    </Text>

                    <View style={[styles.radiusMask]} />
                </View>
                <View style={styles.textContainer}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Detail pemesanan ${this.props.data.subtitle}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Detail Transaksi
                    </Text>
                    </TouchableOpacity>

                    </TouchableWithoutFeedback>
                </View>
                <View style={styles.textContainerButton}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Lacak status pemesanan ${this.props.data.subtitle}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Lacak Status
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableOpacity>
        );
    }
}
