// react-native-gifted-chat/src/components/SliderEntry.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, TouchableWithoutFeedback, InlineImage, Dimensions, Platform, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { ParallaxImage } from 'react-native-snap-carousel';
import styles from '/Users/catherinepricillas/Chatbot/node_modules/react-native-gifted-chat/src/styles/SliderEntry.style';

export default class SliderEntry extends Component {

    static propTypes = {
        data: PropTypes.object.isRequired,
        onPress: PropTypes.func.isRequired,
        onPress2: PropTypes.func.isRequired,
        even: PropTypes.bool,
        parallax: PropTypes.bool,
        parallaxProps: PropTypes.object
    };


    get image () {
        const { data: { illustration }, parallax, parallaxProps, even } = this.props;
        const IS_IOS = Platform.OS === 'ios';
        const { width: viewportWidth, height: viewportHeight } = Dimensions.get('window');

        function wp (percentage) {
            const value = (percentage * viewportWidth) / 100;
            return Math.round(value);
        }

        const slideHeight = viewportHeight * 0.4;
        const slideWidth = wp(72);
        const itemHorizontalMargin = wp(0.17);

        const sliderWidth = viewportWidth;
        const itemWidth = slideWidth + itemHorizontalMargin * 0.00000002;

        const entryBorderRadius = 8;
        return parallax ? (
            <ParallaxImage
              source={illustration}
              containerStyle={{flex: 1,
                marginBottom: IS_IOS ? 0 : -1, // Prevent a random Android rendering issue
                backgroundColor: 'white',
                borderTopLeftRadius: entryBorderRadius,
                borderTopRightRadius: entryBorderRadius,
                borderWidth: StyleSheet.hairlineWidth,
                borderColor: '#eba43b'}}
              style={{...StyleSheet.absoluteFillObject,
                resizeMode: 'cover',
                borderRadius: IS_IOS ? entryBorderRadius : 0,
                borderTopLeftRadius: entryBorderRadius,
                borderTopRightRadius: entryBorderRadius}}
                parallaxFactor={0}
              showSpinner={true}
              spinnerColor={even ? 'rgba(255, 255, 255, 0.4)' : 'rgba(0, 0, 0, 0.25)'}
              {...parallaxProps}
            />
        ) : (
            <Image
              source={illustration}
              style={{...StyleSheet.absoluteFillObject,
        resizeMode: 'cover',
        borderRadius: IS_IOS ? entryBorderRadius : 0,
        borderTopLeftRadius: entryBorderRadius,
        borderTopRightRadius: entryBorderRadius}}
            />
        );
    }


    render () {
        const IS_IOS = Platform.OS === 'ios';
        const { width: viewportWidth, height: viewportHeight } = Dimensions.get('window');

        function wp (percentage) {
            const value = (percentage * viewportWidth) / 100;
            return Math.round(value);
        }

        const slideHeight = viewportHeight * 0.4;
        const slideWidth = wp(72);
        const itemHorizontalMargin = wp(0.17);

        const sliderWidth = viewportWidth;
        const itemWidth = slideWidth + itemHorizontalMargin * 0.00000002;

        const entryBorderRadius = 8;
        const { data: { title, subtitle }, even } = this.props;

        const uppercaseTitle = title ? (
            <Text
              style={
                {
                    color: '#1a1917',
                    fontSize: 13,
                    fontWeight: 'bold',
                    letterSpacing: 0.5
                }
              }
              numberOfLines={2}
            >
                { title }
            </Text>
        ) : false;

        return (
            <TouchableOpacity
              activeOpacity={1}
              style={{
                width: itemWidth,
                height: slideHeight,
                paddingHorizontal: itemHorizontalMargin,
                paddingBottom: 10
              }}
              onPress={() => { alert(`You've clicked '${title}'`); }}
              >
                <View style={{
                    position: 'absolute',
                    top: 0,
                    left: itemHorizontalMargin,
                    right: itemHorizontalMargin,
                    bottom: 18,
                    shadowColor: '#1a1917',
                    shadowOpacity: 0.25,
                    shadowOffset: { width: 0, height: 10 },
                    shadowRadius: 10,
                    borderRadius: entryBorderRadius
                }} />
                <View style={{
                    flex: 1,
                    marginBottom: IS_IOS ? 0 : -1, // Prevent a random Android rendering issue
                    backgroundColor: 'white',
                    borderTopLeftRadius: entryBorderRadius,
                    borderTopRightRadius: entryBorderRadius,
                    borderWidth: StyleSheet.hairlineWidth,
                    borderColor: '#eba43b'
                }} >
                    { this.image }
                    <View style={{
                        position: 'absolute',
                        bottom: 0,
                        left: 0,
                        right: 0,
                        height: entryBorderRadius,
                        backgroundColor: 'white'
                    }} />
                </View>
                <View style={{
                    justifyContent: 'center',
                    paddingTop: 15 - entryBorderRadius,
                    paddingBottom: 10,
                    paddingHorizontal: 16,
                    backgroundColor: 'white',
                    borderLeftWidth: StyleSheet.hairlineWidth,
                    borderBottomWidth: StyleSheet.hairlineWidth,
                    borderRightWidth: StyleSheet.hairlineWidth,
                    borderColor: '#eba43b'}}>
                    { uppercaseTitle }
                    <Text
                      style={{
                        marginTop: 6,
                        color: '#888888',
                        fontSize: 12,
                      }}
                      numberOfLines={2}
                    >
                        { subtitle }{'                               '}
                    <Image style={{height: 14, width: 60}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/five-stars.png')}
                      />
                    </Text>
                  
                </View>
                <View style={{
                    justifyContent: 'center',
                    paddingTop: 15 - entryBorderRadius,
                    paddingBottom: 10,
                    paddingHorizontal: 16,
                    backgroundColor: 'white',
                    borderLeftWidth: StyleSheet.hairlineWidth,
                    borderBottomWidth: StyleSheet.hairlineWidth,
                    borderRightWidth: StyleSheet.hairlineWidth,
                    borderColor: '#eba43b'}}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={() => { alert(`Hai`); }}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Lihat Detail Produk
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
                <View style={{justifyContent: 'center',
                    paddingTop: 15 - entryBorderRadius,
                    paddingBottom: 10,
                    paddingHorizontal: 16,
                    backgroundColor: 'white',
                    borderLeftWidth: StyleSheet.hairlineWidth,
                    borderBottomWidth: StyleSheet.hairlineWidth,
                    borderRightWidth: StyleSheet.hairlineWidth,
                    borderColor: '#eba43b'}}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={this.onPressChild2}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Produk Serupa
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
                <View style={{
                justifyContent: 'center',
                paddingTop: 15 - entryBorderRadius,
                paddingBottom: 10,
                paddingHorizontal: 16,
                backgroundColor: 'white',
                borderBottomLeftRadius: entryBorderRadius,
                borderBottomRightRadius: entryBorderRadius,
                borderLeftWidth: StyleSheet.hairlineWidth,
                borderBottomWidth: StyleSheet.hairlineWidth,
                borderRightWidth: StyleSheet.hairlineWidth,
                borderColor: '#eba43b'}}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={() => { alert(`Hai`); }}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Beli Produk
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableOpacity>
        );
    }
}
