// react-native-gifted-chat/src/components/SliderEntry5.js
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, TouchableWithoutFeedback, InlineImage } from 'react-native';
import PropTypes from 'prop-types';
import { ParallaxImage } from 'react-native-snap-carousel';
import styles from '/Users/catherinepricillas/Chatbot/node_modules/react-native-gifted-chat/src/styles/Address.style';

export default class SliderEntry5 extends Component {

      constructor(props){
        super(props);
      }

    static propTypes = {
        data: PropTypes.object.isRequired,
        onPress: PropTypes.func.isRequired,
        even: PropTypes.bool,
        parallax: PropTypes.bool,
        parallaxProps: PropTypes.object
    };

  _onPress(text2) {
      this.props.onPress(text2);
  }

  _onPress2() {
      this.props.onPress2();
  }

  _onPress3(text2, idx) {
      this.props.onPress3(text2, idx);
  }


    get image () {
        const { data: { illustration }, parallax, parallaxProps, even, idx } = this.props;

        return parallax ? (
            <ParallaxImage
              source={illustration}
              containerStyle={[styles.imageContainer, even ? styles.imageContainerEven : {}]}
              style={styles.image}
              parallaxFactor={0}
              showSpinner={true}
              spinnerColor={even ? 'rgba(255, 255, 255, 0.4)' : 'rgba(0, 0, 0, 0.25)'}
              {...parallaxProps}
            />
        ) : (
            <Image
              source={illustration}
              style={styles.image}
            />
        );
    }


    render () {
        const { data: { title, subtitle }, even, idx } = this.props;

        const uppercaseTitle = title ? (
            <Text
              style={[styles.title, even ? styles.titleEven : {}]}
              numberOfLines={2}
            >
                { title }
            </Text>
        ) : false;


        if (this.props.data.next === true) {
          return (
            <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              onPress={()=>this._onPress(`Tambah alamat baru`)}
              >
                <View style={styles.shadow} />
                <View style={styles.imageContainer} >
                    <Image
                      source={this.props.data.illustration}
                      style={styles.image2}
                    />
                    <View style={styles.radiusMask} />
                </View>
                <View style={styles.textContainerButton}>
                </View>
            </TouchableOpacity>
          );
        }

        return (
            <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              onPress={()=>this._onPress(`Pilih Alamat ${this.props.data.namaalamat}`)}
              >
                <View style={styles.shadow} />
                <View style={[styles.imageContainer]} >
                    <Text
                      style={styles.titleEven}
                    >
                      Alamat {this.props.data.namaalamat}
                    </Text>
                    <Text
                      style={styles.title}
                    >
                      Nama: {this.props.data.nama}
                    </Text>
                    <Text
                      style={styles.title}
                    >
                      Alamat: {this.props.data.alamat}
                    </Text>
                                        <Text
                      style={styles.title}
                    >
                      Nomor HP: {this.props.data.no}
                    </Text>

                    <View style={[styles.radiusMask]} />
                </View>
                <View style={styles.textContainer}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Pilih Alamat ${this.props.data.namaalamat}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Pilih Alamat
                    </Text>
                    </TouchableOpacity>

                    </TouchableWithoutFeedback>
                </View>
                <View style={styles.textContainerButton}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress(`Edit Alamat ${this.props.data.namaalamat}`)}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Edit Alamat
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableOpacity>
        );
    }
}
