// react-native-gifted-chat/src/Message.js
/* eslint no-use-before-define: ["error", { "variables": false }], react-native/no-inline-styles: 0 */

import PropTypes from 'prop-types';
import React from 'react';
import { View, ViewPropTypes, StyleSheet, SafeAreaView } from 'react-native';

import Avatar from './Avatar';
import Bubble from './Bubble';
import SystemMessage from './SystemMessage';
import Day from './Day';
import ScrollCategories from './ScrollCategories';
import ScrollCatalog from './ScrollCatalog';
import PayButton from './PayButton';
import Basket from './Basket';
import Wishlist from './Wishlist';
import History from './History';

import { isSameUser, isSameDay } from './utils';

export default class Message extends React.Component {

  getInnerComponentProps() {
    const { containerStyle, ...props } = this.props;
    return {
      ...props,
      isSameUser,
      isSameDay,
    };
  }

  renderDay() {
    if (this.props.currentMessage.createdAt) {
      const dayProps = this.getInnerComponentProps();
      if (this.props.renderDay) {
        return this.props.renderDay(dayProps);
      }
      return <Day {...dayProps} />;
    }
    return null;
  }

  renderBubble() {
    const bubbleProps = this.getInnerComponentProps();
    if (this.props.renderBubble) {
      return this.props.renderBubble(bubbleProps);
    }
    return <Bubble {...bubbleProps} />;
  }

  renderSystemMessage() {
    const systemMessageProps = this.getInnerComponentProps();
    if (this.props.renderSystemMessage) {
      return this.props.renderSystemMessage(systemMessageProps);
    }
    return <SystemMessage {...systemMessageProps} />;
  }


  renderAvatar() {
    if (this.props.user._id === this.props.currentMessage.user._id && !this.props.showUserAvatar) {
      return null;
    }
    const avatarProps = this.getInnerComponentProps();
    const { currentMessage } = avatarProps;
    if (currentMessage.user.avatar === null) {
      return null;
    }
    return <Avatar {...avatarProps} />;
  }

  renderCategories() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const categoriesProps = this.getInnerComponentProps();
    if (this.props.renderCategories) {
      return this.props.renderCategories(categoriesProps);
    }
    return <ScrollCategories {...categoriesProps} />;
  }

  renderCatalog() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const catalogProps = this.getInnerComponentProps();
    if (this.props.renderCatalog) {
      return this.props.renderCatalog(catalogProps);
    }
    return <ScrollCatalog {...catalogProps} />;
  }

  renderBasket() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const basketProps = this.getInnerComponentProps();
    if (this.props.renderBasket) {
      return this.props.renderBasket(basketProps);
    }
    return <Basket {...basketProps} />;
  }

  renderHistory() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const historyProps = this.getInnerComponentProps();
    if (this.props.renderHistory) {
      return this.props.renderHistory(historyProps);
    }
    return <History {...historyProps} />;
  }

  renderWishlist() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const wishlistProps = this.getInnerComponentProps();
    if (this.props.renderWishlist) {
      return this.props.renderWishlist(wishlistProps);
    }
    return <Wishlist {...wishlistProps} />;
  }


  renderPayButton() {

    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    if (sameUser) {
      return null;
    }
    const payProps = this.getInnerComponentProps();
    if (this.props.renderPayButton) {
      return this.props.renderPayButton(payProps);
    }
    return <PayButton {...payProps} />;
  }

  render() {
    const sameUser = isSameUser(this.props.currentMessage, this.props.nextMessage);
    return (
      <SafeAreaView>
        {this.renderDay()}
        {this.props.currentMessage.system ? (
          this.renderSystemMessage()
        ) : (
          <View
            style={[
              styles[this.props.position].container,
              { marginBottom: sameUser ? 2 : 10 },
              !this.props.inverted && { marginBottom: 2 },
              this.props.containerStyle[this.props.position],
            ]}
          >
            {this.props.position === 'left' ? this.renderAvatar() : null}
            {this.renderBubble()}
            {this.props.position === 'right' ? this.renderAvatar() : null}
          </View>
        )}
        {this.props.position === 'right' ? null : this.renderCategories()}
        {this.props.position === 'right' ? null : this.renderCatalog()}
        {this.props.position === 'right' ? null : this.renderBasket()}
        {this.props.position === 'right' ? null : this.renderHistory()}
        {this.props.position === 'right' ? null : this.renderWishlist()}
        {this.props.position === 'right' ? null : this.renderPayButton()}
      </SafeAreaView>
    );
  }

}

const styles = {
  left: StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'flex-end',
      justifyContent: 'flex-start',
      marginLeft: 8,
      marginRight: 0,
    },
  }),
  right: StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'flex-end',
      justifyContent: 'flex-end',
      marginLeft: 0,
      marginRight: 8,
    },
  }),
};

Message.defaultProps = {
  renderAvatar: undefined,
  renderBubble: null,
  renderDay: null,
  renderCategories:null,
  renderCatalog:null,
  renderBasket:null,
  renderHistory:null,
  renderWishlist:null,
  renderPayButton:null,
  renderSystemMessage: null,
  position: 'left',
  currentMessage: {},
  nextMessage: {},
  previousMessage: {},
  user: {},
  containerStyle: {},
  showUserAvatar: true,
  inverted: true,
};

Message.propTypes = {
  renderAvatar: PropTypes.func,
  showUserAvatar: PropTypes.bool,
  renderBubble: PropTypes.func,
  renderCategories: PropTypes.func,
  renderPayButton: PropTypes.func,
  renderCatalog: PropTypes.func,
  renderBasket: PropTypes.func,
  renderHistory: PropTypes.func,
  renderWishlist: PropTypes.func,
  renderDay: PropTypes.func,
  renderSystemMessage: PropTypes.func,
  position: PropTypes.oneOf(['left', 'right']),
  currentMessage: PropTypes.object,
  nextMessage: PropTypes.object,
  previousMessage: PropTypes.object,
  user: PropTypes.object,
  inverted: PropTypes.bool,
  containerStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
};
