// react-native-gifted-chat/src/PayButton.js
import React, {
  Component
} from 'react';
import ReactNative, {
  Text,
  ListView,
  TouchableOpacity,
  View,
  SafeAreaView,
  TouchableWithoutFeedback,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from '/Users/catherinepricillas/Chatbot/node_modules/react-native-gifted-chat/src/styles/Payment.style';

class PayButton extends Component {
  _onPress() {
      this.props.onPress();
  }

  render() {
    if (this.props.isPayment === true) {
    return (
        <SafeAreaView style={{backgroundColor: '#fff'}}>
          <TouchableOpacity
              activeOpacity={1}
              style={styles.slideInnerContainer}
              onPress={()=>this._onPress()}
              >
                <View style={styles.shadow} />
                <View style={styles.imageContainer} >
                    <Text
                      style={styles.title}
                    >
                      Order: #1919292
                    </Text>
                    <Text
                      style={styles.title}
                    >
                      Total: Rp 611.000
                    </Text>
                    <View style={styles.radiusMask} />
                </View>
                <View style={styles.textContainerButton}>
                    <TouchableWithoutFeedback>
                    <TouchableOpacity onPress={()=>this._onPress()}>
                    <Text style={{textAlign: 'center', color: '#eba43b'}}>
                        Bayar Sekarang
                    </Text>
                    </TouchableOpacity>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableOpacity>
        </SafeAreaView>
    )
    }
    return null;
  }
}

PayButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  isPayment: PropTypes.bool,
  onPayment: PropTypes.func,
}

PayButton.defaultProps = {
  isQuickReply: false,
}

module.exports = PayButton;
