// react-native-gifted-chat/src/PersistentMenu.js
import React, {
  Component
} from 'react';
import ReactNative, {
  Text,
  ListView,
  TouchableOpacity,
  View,
  SafeAreaView,
  Image,
} from 'react-native';
import PropTypes from 'prop-types';

class PersistentMenu extends Component {
  _onPress(data) {
      this.props.onPress(data);
  }

  render() {
    if (this.props.isMenu === true) {
    return (
        <SafeAreaView style={{backgroundColor: '#6a9cde'}}>
          <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress('Mulai Pencarian Baru')}>
              <View style={
                {
                  backgroundColor: "#fff",
                  borderColor: "#6a9cde",
                  paddingTop:this.props.paddingItemTop,
                  paddingBottom:this.props.paddingItemBottom,
                  paddingLeft:this.props.paddingItemLeft,
                  paddingRight:this.props.paddingItemRight,
                  borderRadius:this.props.borderItemRadius,
                  borderTopWidth:this.props.borderItemTopWidth,
                  borderLeftWidth:this.props.borderItemLeftWidth,
                  borderRightWidth:this.props.borderItemRightWidth,
                  marginRight: this.props.marginItemRight
                }
              }>
                <Text style={
                  {
                    color:"#e0a551",
                    fontSize:this.props.fontSize,
                    fontWeight:this.props.fontWeight}
                  }>
                  <Image style={{height: 18, width: 18}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/refresh.png')}
                      />
                      {'   '}
                  Mulai Pencarian Baru</Text>
              </View>
          </TouchableOpacity>
          <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress('Keranjang Belanja')}>
              <View style={
                {
                  backgroundColor: "#fff",
                  borderColor: "#6a9cde",
                  paddingTop:this.props.paddingItemTop,
                  paddingBottom:this.props.paddingItemBottom,
                  paddingLeft:14,
                  paddingRight:this.props.paddingItemRight,
                  borderRadius:this.props.borderItemRadius,
                  borderTopWidth:this.props.borderItemTopWidth,
                  borderLeftWidth:this.props.borderItemLeftWidth,
                  borderRightWidth:this.props.borderItemRightWidth,
                  marginRight: this.props.marginItemRight
                }
              }>
                <Text style={
                  {
                    color:"#e0a551",
                    fontSize:this.props.fontSize,
                    fontWeight:this.props.fontWeight}
                  }>
                  <Image style={{height: 18, width: 20}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/cart.png')}
                      />
                      {'   '}Keranjang Belanja</Text>
              </View>
          </TouchableOpacity>
          <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress('Wishlist')}>
              <View style={
                {
                  backgroundColor: "#fff",
                  borderColor: "#6a9cde",
                  paddingTop:this.props.paddingItemTop,
                  paddingBottom:this.props.paddingItemBottom,
                  paddingLeft:14,
                  paddingRight:this.props.paddingItemRight,
                  borderRadius:this.props.borderItemRadius,
                  borderTopWidth:this.props.borderItemTopWidth,
                  borderLeftWidth:this.props.borderItemLeftWidth,
                  borderRightWidth:this.props.borderItemRightWidth,
                  marginRight: this.props.marginItemRight
                }
              }>
                <Text style={
                  {
                    color:"#e0a551",
                    textAlign : "center",
                    fontSize:this.props.fontSize,
                    fontWeight:this.props.fontWeight}
                  }>
                  <Image style={{height: 18, width: 20}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/heartlined.png')}
                      />
                      {'   '}Wishlist</Text>
              </View>
          </TouchableOpacity>
          <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress('Histori Transaksi')}>
              <View style={
                {
                  backgroundColor: "#fff",
                  borderColor: "#6a9cde",
                  paddingTop:this.props.paddingItemTop,
                  paddingBottom:this.props.paddingItemBottom,
                  paddingLeft:14,
                  paddingRight:this.props.paddingItemRight,
                  borderRadius:this.props.borderItemRadius,
                  borderTopWidth:this.props.borderItemTopWidth,
                  borderLeftWidth:this.props.borderItemLeftWidth,
                  borderRightWidth:this.props.borderItemRightWidth,
                  marginRight: this.props.marginItemRight
                }
              }>
                <Text style={
                  {
                    color:"#e0a551",
                    textAlign : "center",
                    fontSize:this.props.fontSize,
                    fontWeight:this.props.fontWeight}
                  }>
                  <Image style={{height: 20, width: 20}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/history.png')}
                      />
                      {'   '}Histori Transaksi</Text>
              </View>
          </TouchableOpacity>
          <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress('Bantuan')}>
              <View style={
                {
                  backgroundColor: "#fff",
                  borderColor: "#6a9cde",
                  paddingTop:this.props.paddingItemTop,
                  paddingBottom:this.props.paddingItemBottom,
                  paddingLeft:this.props.paddingItemLeft,
                  paddingRight:this.props.paddingItemRight,
                  borderTopWidth:this.props.borderItemTopWidth,
                  borderLeftWidth:this.props.borderItemLeftWidth,
                  borderRightWidth:this.props.borderItemRightWidth,
                }
              }>
                <Text style={
                  {
                    color:"#e0a551",
                    fontSize:this.props.fontSize,
                    fontWeight:this.props.fontWeight}
                  }>
                  <Image style={{height: 18, width: 18}}
                        source={require('/Users/catherinepricillas/Chatbot/shared/static/help.png')}
                      />
                      {'   '}Bantuan</Text>
              </View>
          </TouchableOpacity>
        </SafeAreaView>
    )
    }
    return null;
  }
}

PersistentMenu.propTypes = {
  items: PropTypes.array.isRequired,
  onPress: PropTypes.func.isRequired,
  selectedOpacity: PropTypes.number,
  paddingItemTop: PropTypes.number,
  paddingItemBottom: PropTypes.number,
  paddingItemLeft: PropTypes.number,
  paddingItemRight: PropTypes.number,
  marginItemRight: PropTypes.number,
  borderItemLeftWidth: PropTypes.number,
  borderItemTopWidth: PropTypes.number,
  borderItemRightWidth: PropTypes.number,
  borderItemRadius: PropTypes.number,
  fontSize: PropTypes.number,
  paddingListEnd: PropTypes.number,
  upperCase: PropTypes.bool,
  fontWeight: PropTypes.string,
  isMenu: PropTypes.bool,
  onMenu: PropTypes.func,
}

PersistentMenu.defaultProps = {
  selectedOpacity: 0.7,
  paddingItemTop: 7,
  paddingItemBottom: 7,
  paddingItemLeft: 16,
  paddingItemRight: 16,
  borderItemTopWidth: 1,
  borderItemLeftWidth: 1,
  borderItemRightWidth: 1,
  fontSize: 15.5,
  paddingListEnd: 7,
  upperCase: false,
  isMenu: false,
}

module.exports = PersistentMenu;
