// react-native-gifted-chat/src/ScrollingButtonMenu.js
import React, {
  Component
} from 'react';
import ReactNative, {
  Text,
  ListView,
  TouchableOpacity,
  View,
  SafeAreaView,
} from 'react-native';
import PropTypes from 'prop-types';

var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
class ScrollingButtonMenu extends Component {
  _onPress(data) {
      this.props.onPress(data);
  }

  _renderRow(data,idx) {
      return(
        <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress(data)} key={idx}>
            <View style={
              {
                backgroundColor:data.backgroundColor,
                borderColor:data.borderColor,
                paddingTop:this.props.paddingItemTop,
                paddingBottom:this.props.paddingItemBottom,
                paddingLeft:this.props.paddingItemLeft,
                paddingRight:this.props.paddingItemRight,
                borderRadius:this.props.borderItemRadius,
                borderWidth:this.props.borderItemWidth,
                marginRight: this.props.marginItemRight
              }
            }>
              <Text style={
                {
                  color:data.textColor,
                  fontSize:this.props.fontSize,
                  fontWeight:this.props.fontWeight}
                }>{this.props.upperCase ? data.text.toUpperCase() : data.text}</Text>
            </View>
        </TouchableOpacity>
      );
  }
  render() {
    if (this.props.isQuickReply === true) {
    return (
        <SafeAreaView style={{backgroundColor: '#fff'}}>
          <ListView
              dataSource={ds.cloneWithRows(this.props.items)}
              enableEmptySections={true}
              showsHorizontalScrollIndicator={false}
              horizontal={true}
              style={this.props.style}
              renderFooter={()=>{
                  return (<View style={{padding:this.props.paddingListEnd}}/>)
              }}
              renderRow={
                (data) => this._renderRow(data)
              }
          />
        </SafeAreaView>
    )
    }
    return null;
  }
}

ScrollingButtonMenu.propTypes = {
  items: PropTypes.array.isRequired,
  onPress: PropTypes.func.isRequired,
  selectedOpacity: PropTypes.number,
  paddingItemTop: PropTypes.number,
  paddingItemBottom: PropTypes.number,
  paddingItemLeft: PropTypes.number,
  paddingItemRight: PropTypes.number,
  marginItemRight: PropTypes.number,
  borderItemWidth: PropTypes.number,
  borderItemRadius: PropTypes.number,
  fontSize: PropTypes.number,
  paddingListEnd: PropTypes.number,
  upperCase: PropTypes.bool,
  fontWeight: PropTypes.string,
  isQuickReply: PropTypes.bool,
  onQuickReply: PropTypes.func,
}

ScrollingButtonMenu.defaultProps = {
  selectedOpacity: 0.7,
  paddingItemTop: 9,
  paddingItemBottom: 9,
  paddingItemLeft: 12,
  paddingItemRight: 12,
  marginItemRight: 7,
  borderItemWidth: 0.3,
  borderItemRadius: 25,
  fontSize: 14,
  paddingListEnd: 7,
  upperCase: false,
  isQuickReply: false,
}

module.exports = ScrollingButtonMenu;
