// react-native-gifted-chat/src/ScrollCategories.js
import React, {
  Component
} from 'react';
import ReactNative, {
  Text,
  ListView,
  TouchableOpacity,
  View,
  SafeAreaView,
  Image
} from 'react-native';
import PropTypes from 'prop-types';
import Lightbox from 'react-native-lightbox';

var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
class ScrollCategories extends Component {
  _onPress(data) {
      this.props.onPress(data);
  }

  _renderRow(data,idx) {
      return(
        <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress(data)} key={idx}>
            <View style={
              {
                backgroundColor:data.backgroundColor,
                borderColor:data.borderColor,
                paddingTop:this.props.paddingItemTop,
                paddingBottom:this.props.paddingItemBottom,
                paddingLeft:this.props.paddingItemLeft,
                paddingRight:this.props.paddingItemRight,
                borderRadius:this.props.borderItemRadius,
                borderWidth:this.props.borderItemWidth,
                marginRight: this.props.marginItemRight,
                width: 152,
                height: 152,
              }
            }>
                <Image source={data.url}/>
            </View>
        </TouchableOpacity>
      );
  }
  render() {
    if (this.props.isCategories === true) {
    return (
        <SafeAreaView>
          <ListView
              dataSource={ds.cloneWithRows(this.props.items2)}
              enableEmptySections={true}
              showsHorizontalScrollIndicator={false}
              horizontal={true}
              style={this.props.style}
              renderFooter={()=>{
                  return (<View style={{padding:this.props.paddingListEnd}}/>)
              }}
              renderRow={
                (data) => this._renderRow(data)
              }
          />
        </SafeAreaView>
    )
    }
    return null;
  }
}

ScrollCategories.propTypes = {
  items2: PropTypes.array.isRequired,
  onPress: PropTypes.func.isRequired,
  selectedOpacity: PropTypes.number,
  paddingItemTop: PropTypes.number,
  paddingItemBottom: PropTypes.number,
  paddingItemLeft: PropTypes.number,
  paddingItemRight: PropTypes.number,
  marginItemRight: PropTypes.number,
  borderItemWidth: PropTypes.number,
  borderItemRadius: PropTypes.number,
  fontSize: PropTypes.number,
  paddingListEnd: PropTypes.number,
  upperCase: PropTypes.bool,
  fontWeight: PropTypes.string,
  isCategories: PropTypes.bool,
  onCategories: PropTypes.func,
}

ScrollCategories.defaultProps = {
  selectedOpacity: 0.7,
  borderColor: '#eba43b',
  paddingItemTop: -1,
  paddingItemBottom: 0,
  paddingItemLeft: -1,
  paddingItemRight: 0,
  marginItemRight: 7,
  borderItemWidth: 1,
  borderItemRadius: 0,
  fontSize: 14,
  paddingListEnd: 7,
  upperCase: false,
  isCate: false,
}

module.exports = ScrollCategories;
