// react-native-gifted-chat/src/ScrollCategories.js
import React, {
  Component
} from 'react';
import ReactNative, {
  Text,
  ListView,
  TouchableOpacity,
  View,
  SafeAreaView,
  Image
} from 'react-native';
import PropTypes from 'prop-types';
import Lightbox from 'react-native-lightbox';

var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
class ScrollCategories extends Component {
  _onPress(data) {
      this.props.onPress(data);
  }

  _renderRow(data,idx) {
      return(
        <TouchableOpacity activeOpacity={this.props.selectedOpacity} onPress={()=>this._onPress(data)} key={idx}>
          <View 
            <Lightbox
            {...lightboxProps}
            >
              <Image
                {...imageProps}
                source={'./shared/static/present.jpg'}
              />
            </Lightbox>
          </View>
        </TouchableOpacity>
      );
  }
  render() {
    if (this.props.isQuickReply === true) {
    return (
        <SafeAreaView>
          <ListView
              dataSource={ds.cloneWithRows(this.props.items)}
              enableEmptySections={true}
              showsHorizontalScrollIndicator={false}
              horizontal={true}
              style={this.props.style}
              renderFooter={()=>{
                  return (<View style={{padding:this.props.paddingListEnd}}/>)
              }}
              renderRow={
                (data) => this._renderRow(data)
              }
          />
        </SafeAreaView>
    )
    }
    return null;
  }
}

ScrollCategories.propTypes = {
  items: PropTypes.array.isRequired,
  onPress: PropTypes.func.isRequired,
  selectedOpacity: PropTypes.number,
  paddingItemTop: PropTypes.number,
  paddingItemBottom: PropTypes.number,
  paddingItemLeft: PropTypes.number,
  paddingItemRight: PropTypes.number,
  marginItemRight: PropTypes.number,
  borderItemWidth: PropTypes.number,
  borderItemRadius: PropTypes.number,
  fontSize: PropTypes.number,
  paddingListEnd: PropTypes.number,
  upperCase: PropTypes.bool,
  fontWeight: PropTypes.string,
  isQuickReply: PropTypes.bool,
  onQuickReply: PropTypes.func,
}

ScrollCategories.defaultProps = {
  selectedOpacity: 0.7,
  paddingItemTop: 7,
  paddingItemBottom: 7,
  paddingItemLeft: 16,
  paddingItemRight: 16,
  marginItemRight: 7,
  borderItemWidth: 1,
  borderItemRadius: 25,
  fontSize: 14,
  paddingListEnd: 7,
  upperCase: false,
  isQuickReply: false,
}

module.exports = ScrollCategories;
